libs/beacon/src/beacon_setup_service.cpp

```diff
@@ -128,238 +169,772 @@ BeaconSetupService::State BeaconSetupService::OnIdle()
 BeaconSetupService::State BeaconSetupService::OnWaitForDirectConnections()
 {
   std::lock_guard<std::mutex> lock(mutex_);
-  auto                        v_peers = endpoint_.GetDirectlyConnectedPeers();
-  std::unordered_set<Address> peers(v_peers.begin(), v_peers.end());
+  dkg_state_gauge_->set(static_cast<uint8_t>(State::WAIT_FOR_DIRECT_CONNECTIONS));
 
-  bool     all_connected = true;
-  uint16_t connected     = 0;
+  auto                              v_peers = endpoint_.GetDirectlyConnectedPeers();
+  std::unordered_set<MuddleAddress> peers(v_peers.begin(), v_peers.end());
+
+  std::set<MuddleAddress> connected;
 
   for (auto &m : beacon_->aeon.members)
   {
     // Skipping own address
     if (m == identity_)
     {
+      connected.insert(m.identifier());
       continue;
     }
 
     // Checking if other peers are there
     if (peers.find(m.identifier()) == peers.end())
     {
-      all_connected = false;
-
       // TODO(tfr): Request muddle to connect.
     }
     else
     {
-      connected++;
+      connected.insert(m.identifier());
     }
   }
 
-  if (all_connected)
+  if (connected.size() == beacon_->aeon.members.size())
```

Comment:
doesn't this want to be -1 since you don't count yourself as a connection?

Reply:
I inserted the nodes own identity into connected so it is correct as it is, however I will change it so that its own identity is not inserted and we have the -1.